src/perl/lib/WTSI/NPG/Genotyping/VCF/Slurper.pm

```diff
@@ -0,0 +1,145 @@
+use utf8;
+
+package WTSI::NPG::Genotyping::VCF::Slurper;
+
+use Moose;
+
+use File::Temp qw /tempdir/;
+use WTSI::NPG::iRODS;
+use WTSI::NPG::iRODS::DataObject;
+use WTSI::NPG::Genotyping::VCF::DataRowParser;
+use WTSI::NPG::Genotyping::VCF::HeaderParser;
+
+with 'WTSI::DNAP::Utilities::Loggable';
+
+has 'input_path'  =>
+   (is            => 'ro',
+    isa           => 'Str',
+    required      => 1,
+    documentation => "Input path in iRODS or local filesystem,".
+                     " or '-' for STDIN.",
+   );
+
+has 'irods'       =>
+   (is            => 'ro',
+    isa           => 'WTSI::NPG::iRODS',
+    documentation => '(Optional) iRODS instance from which to read data. '.
+                     'If not given, input (if any) is assumed to be a '.
+                     'path on the local filesystem.',
+   );
+
```

Comment:
What is expected to happen when I pass an 'input_filehandle' initarg to the constructor?